# use katex in jupyterlab  
pip install jupyterlab-katex

# Bayes' Rule  
$P(A|B) = \frac{P(B|A) \cdot P(A)}{P(B)}$

# Law of Total Probability  
$P(B) = \displaystyle\sum_{i}^{} P(A_i \cap B) = \displaystyle\sum_{i}^{} P(B|A_i)\cdot P(A_i)$

# Condition  
$P(B|A) = \frac{800}{1000}$  
$P(A^c\cap B) + P(A \cap B^c) = \frac{100}{1000}$  
$P(A) = \frac{50}{10000}$  

# Solution
$P(A^c\cap B) + P(A \cap B^c) = \frac{100}{1000}$  
$P(B|A^c)\cdot P(A^c) + P(B^c|A)\cdot P(A) = \frac{100}{1000}$  
$P(B|A^c) = \{\frac{100}{1000} - \frac{50}{10000}\cdot(1-\frac{800}{1000})\}\div (1-\frac{50}{10000})$  
  
$P(A|B) = \frac{P(B|A)\cdot P(A)}{P(B)}$  
$P(A|B) = \frac{P(B|A)\cdot P(A)}{P(A\cap B)+P(A^c \cap B)}$  
$P(A|B) = \frac{P(B|A)\cdot P(A)}{P(B|A)\cdot P(A)+P(B|A^c)\cdot P(A^c)}$  
$P(A|B) = \frac{P(B|A)\cdot P(A)}{P(B|A)\cdot P(A)+P(B|A^c)\cdot P(A^c)}$  


In [6]:
p_ba = 800/1000
p_a = 50/10000
p_bac = (100/1000 - p_a*(1-p_ba))/(1-p_a)

print("P(A|B) = ",p_ba*p_a/(p_ba*p_a+p_bac*(1-p_a)))
print("P(B) = ",p_ba*p_a+p_bac*(1-p_a))

P(A|B) =  0.038834951456310676
P(B) =  0.10300000000000001


양성 진단을 내렸을 때 암 발병확률이 낮은 이유는  
암 발병확률(P(A) = 50/10000)에 비해 양성 진단 확률(P(B) = 0.10300000000000001)이 높기 때문이다.  